twisted/test/proto_helpers.py

```diff
@@ -142,6 +142,13 @@ class StringTransport:
     @ivar io: A L{io.BytesIO} which holds the data which has been written to
         this transport since the last call to L{clear}.  Use L{value} instead
         of accessing this directly.
+
+    @ivar lenient: A L{bool} that controls whether L{StringTransport} will
```

Comment:
OK.  It might be perhaps useful to include a sentence like "This enforces `resumeProducing` not be called after the connection is lost by default in order to ensure that any code which does so is not blindly expecting the `resumeProducing` call to have an impact, even on a closed connection; however, it may be safely set to `False` if you know that this is on purpose, and your code won't hang expecting further traffic or anything like that."  OK maybe that's a little long winded but you get the idea.